apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/IntervalTests.cs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Numerics;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class IntervalTests
+    {
+        private static readonly BigInteger NanonosecondsInASecond = (BigInteger)1_000_000_000;
+        private static readonly BigInteger NanonosecondsInAMillisecond = (BigInteger)1_000_000;
+        private static readonly BigInteger NanonosecondsInAMicrosecond = (BigInteger)1_000;
+
+        [Theory]
```

Comment:
Tests for FromXyz out of bounds? (I think they're only covered by parsing at the moment.)